packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/NotebookSessionController.java

```diff
@@ -260,9 +261,11 @@ ResponseEntity<NotebookSession> cloneNotebookSession(
 				permission
 			);
 
-			final Optional<Project> project = projectService.getProject(projectId);
+			final Project project = projectService
+				.getProject(projectId)
+				.orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, messages.get("asset.not-found")));
```

Comment:
This will get caught by the general catch and it has a different error message, so you need to chain them together, or you need additional catch-clauses.